### PR TITLE
Snippets for handling control add-ins

### PR DESCRIPTION
**Pull request by niku-live**
I started porting our control add-ins to AL and noticed that there is no snippets to help adding control add-ins, so I created 2 snippets taddin (which adds controladdin object) and tusercontrol (which adds add-in control to page).

### CODE REVIEW

File: snippets/controladdin.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            "\tHorizontalShrink = ${4:true};",
+            "\tScripts = ",
+            "\t\t'${5:script1.js}',",
+            "\t\t'${6:script2.js}';",
+            "\tStyleSheets =",
+            "\t\t'${7:style1.css}';",
+            "\tStartupScript = '${5:script1.js}';",
+            "\tRecreateScript = '${5:script1.js}';",
+            "\tRefreshScript = '${5:script1.js}';",
+            "\tevent ${8:MyEvent()};",
```
Comment (qutreson) [inline snippets/controladdin.json]: In order to keep it consistent with the other snippets, can you add an empty line (here, "\t") around the procedures and events?

File: snippets/controladdin.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            "\tprocedure ${9:MyProcedure()};",
+            "}"
+        ],
+        "description": "Snippet to create Control Add-In object"
```
Comment (qutreson) [inline snippets/controladdin.json]: In the other snippets, we only have "Snippet: {Snippet name}" in the description. Could you change it to "Snippet: Control Add-In" to make it consistent?

File: snippets/page.json [hunk 3]
```diff
@@ -60,6 +60,18 @@
             "}"
         ]
     },
+    "Snippet: User Control": {
+        "prefix": "tusercontrol",
+        "body": [
+            "usercontrol(${1:ControlName};${2:AddInName})",
+            "{",
+            "\ttrigger ${3:MyTrigger()}\r",
```
Comment (qutreson) [inline snippets/page.json]: Remove \r

File: snippets/page.json [hunk 4]
```diff
@@ -60,6 +60,18 @@
...
+            "\tbegin",
+            "\tend;",
+            "}"
+        ],
+        "description": "Inserts User Control Code to Page"
```
Comment (qutreson) [inline snippets/page.json]: Change description to "Snippet: Page User Control"

**Comment by qutreson** [comment]
Thank you for your contribution @niku-live, I added a few comments on the snippets you added :)

**Comment by niku-live** [comment]
Thank you for your suggestions, I fixed consistency problems (at least I hope :) ).

**Pull request merged**